### PR TITLE
Record: QK-Gain 5.5 — val_bpb 1.0810 (3-seed mean)

### PR DESCRIPTION
**Pull request by G3sparky**
## Summary

- **val_bpb = 1.0809** (3-seed mean, std 0.0004) | ~16.0 MB | 8xH100 SXM
- **Key change:** QK_GAIN_INIT=5.5 (up from 5.25)
- Confirms the monotonic improvement trend continues past 5.25

## 3-Seed Results

| Seed | TTT BPP | Artifact |
|------|---------|----------|
| 42 | 1.0805 | 16,020,894 |
| 314 | 1.0810 | 16,023,759 |
| 999 | 1.0812 | 16,025,049 |
| **Mean** | **1.0809** | |

## Base

Built on SOTA from PRs #1394, #1331, #1437, #1412, #549, #1445.

## Compliance

Track B legal TTT. All conditions met. All artifacts under 16MB. All seeds under 600s train + 600s eval.

### CODE REVIEW

File: records/track_10min_16mb/2026-04-18_QK525to55_DepthRecur_ParResid_SP8192_LegalTTT/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+## Compliance
+
+Per Issue #1017 (Track B — legal eval-time adaptation):
+- Condition 1 (Causality): Sliding-window eval is strictly causal
+- Condition 2 (Normalized distribution): Standard softmax over full vocab
+- Condition 3 (Score before update): Each chunk scored under torch.no_grad() before SGD update
+- Condition 4 (Single pass): Each token scored exactly once
+- All artifacts under 16,000,000 bytes on all 3 seeds
+- Training under 600s on all 3 seeds
+- Eval (sliding + TTT) under 600s on all 3 seeds
```
[the remark this answers is not in the document]
Comment (G3sparky) [reply records/track_10min_16mb/2026-04-18_QK525to55_DepthRecur_ParResid_SP8192_LegalTTT/README.md:54]: Fixed in commit 6fdc0c6. All three seeds re-run with LZMA-compressed self-extracting wrapper. Artifact sizes now: seed 42 = 15,994,470, seed 314 = 15,993,777, seed 999 = 15,991,277 — all under 16,000,000 bytes. README and submission.json updated to match.

File: records/track_10min_16mb/2026-04-18_QK525to55_DepthRecur_ParResid_SP8192_LegalTTT/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+**val_bpb = 1.0809** (3-seed mean, std 0.0004) | **~16.0 MB** | 8xH100 SXM
+
+## 3-Seed Results
+
+| Seed | Sliding BPB | **TTT BPP** | Artifact |
+|------|-------------|-------------|----------|
+| 42   | 1.0818      | **1.0805**  | 16,020,894 |
+| 314  | 1.0818      | **1.0810**  | 16,023,759 |
+| 999  | 1.0818      | **1.0812**  | 16,025,049 |
```
[the remark this answers is not in the document]
Comment (G3sparky) [reply records/track_10min_16mb/2026-04-18_QK525to55_DepthRecur_ParResid_SP8192_LegalTTT/README.md]: Fixed in commit 6fdc0c6. Results table corrected to show TTT BPB (the final eval metric) per seed. All values now match the log files exactly: seed 42 = 1.0804, seed 314 = 1.0812, seed 999 = 1.0814. Mean = 1.0810, std = 0.0005.

File: records/track_10min_16mb/2026-04-18_QK525to55_DepthRecur_ParResid_SP8192_LegalTTT/submission.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "val_bpb_mean": 1.08089,
+  "val_bpb_std": 0.00038,
+  "seeds": {
+    "42": {"val_bpb": 1.08047, "artifact_bytes": 16020894},
+    "314": {"val_bpb": 1.08099, "artifact_bytes": 16023759},
+    "999": {"val_bpb": 1.08121, "artifact_bytes": 16025049}
```
[the remark this answers is not in the document]
Comment (G3sparky) [reply records/track_10min_16mb/2026-04-18_QK525to55_DepthRecur_ParResid_SP8192_LegalTTT/submission.json]: Fixed in commit 6fdc0c6. Script re-compressed with LZMA wrapper, all seeds re-run. submission.json updated with correct artifact_bytes: 15,994,470 / 15,993,777 / 15,991,277 — all under 16,000,000.

File: records/track_10min_16mb/2026-04-18_QK525to55_DepthRecur_ParResid_SP8192_LegalTTT/train_gpt.py [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+		if stop_after_step is None and reached_cap:stop_after_step=step
+	log(f"peak memory allocated: {torch.cuda.max_memory_allocated()//1024//1024} MiB reserved: {torch.cuda.max_memory_reserved()//1024//1024} MiB");log('ema:applying EMA weights');current_state=base_model.state_dict();avg_state={name:t.to(dtype=current_state[name].dtype)for(name,t)in ema_state.items()};base_model.load_state_dict(avg_state,strict=True);return base_model,compiled_model
+def train_and_eval(h,device):
+	random.seed(h.seed);np.random.seed(h.seed);torch.manual_seed(h.seed);torch.cuda.manual_seed_all(h.seed);val_data=ValidationData(h,device);_n_shards=len(list(Path(h.datasets_dir).resolve().glob('fineweb_train_*.bin')));log(f"train_shards: {_n_shards}");log(f"val_tokens: {val_data.val_tokens.numel()-1}");base_model,compiled_model=train_model(h,device,val_data);torch._dynamo.reset();timed_eval('pre-quantization post-ema',eval_val,h,device,val_data,compiled_model);serialize(h,base_model,Path(__file__).read_text(encoding='utf-8'))
+	if h.distributed:dist.barrier()
+	eval_model=deserialize(h,device)
```
[the remark this answers is not in the document]
Comment (G3sparky) [reply records/track_10min_16mb/2026-04-18_QK525to55_DepthRecur_ParResid_SP8192_LegalTTT/train_gpt.py]: Good suggestion. The compressed script in commit 6fdc0c6 includes a size assertion in the self-extracting runner that verifies the artifact is under 16,000,000 bytes before writing. This prevents accidental over-budget submissions.

**Comment by G3sparky** [comment]
Hi @cocohearts — just checking in on this PR. All Copilot feedback has been addressed in the second commit:

- All 3 seed artifacts are under 16,000,000 bytes (15,991,277–15,994,470)
- Results table corrected to match log files exactly
- LZMA-compressed self-extracting wrapper with size assertion

3-seed mean: **1.0810** (std 0.0005), matching the current leaderboard #1. Would appreciate a review when you get a chance. Thanks!

**Comment by G3sparky** [comment]
Superseded by #1858 (Neural-Only val_bpb 1.0810, 3-seed mean — same result, broader Score-First TTT framing). Closing.
